src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -255,6 +257,66 @@ private static string ReplaceWith24HourClock(string format)
             return Regex.Replace(format, pattern, "HH${nonHours}", RegexOptions.IgnorePatternWhitespace);
         }
 
+        /// <summary>
+        /// Resolves month/minute ambiguity and 24h vs AM/PM.
+        /// </summary>
+        /// <param name="format">Format string.</param>
+        /// <returns></returns>
+        private static string ResolveDateTimeFormatAmbiguities(string format)
+        {
+            if (format.Length == 0)
+            {
+                return format;
+            }
+
+            // Handling especial cases
+            if (format.Length == 1)
+            {
+                switch (format)
+                {
+                    case "m":
+                    case "h":
+                        return $"%{format.ToUpperInvariant()}";
+                    default:
+                        return $"%{format}";
+                }
+            }
+
+            Regex minutesBeforeSecondsRegex = new Regex("[M]+[^dDyYhHmM]+[sS]", RegexOptions.Compiled);
+            Regex minutesAfterHoursRegex = new Regex("[hH][^dDyYmM]+[M]+", RegexOptions.Compiled);
+            Regex time12hRegex = new Regex("[aA][mM]\\/[pP][mM]|[aA]\\/[pP]", RegexOptions.Compiled);
+            Regex ampmReplaceRegex = new Regex("[aA][mM]\\/[pP][mM]", RegexOptions.Compiled);
+            Regex apReplaceRegex = new Regex("[aA]\\/[pP]", RegexOptions.Compiled);
```

Comment:
Move those to static readonly members in the class, so that the regex compilation only happens once per app domain (instead of every call to the Text function)